sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregates.scala

```diff
@@ -17,285 +17,159 @@
 
 package org.apache.spark.sql.catalyst.expressions
 
-import com.clearspring.analytics.stream.cardinality.HyperLogLog
-
 import org.apache.spark.sql.types._
-import org.apache.spark.sql.catalyst.trees
-import org.apache.spark.sql.catalyst.errors.TreeNodeException
-import org.apache.spark.util.collection.OpenHashSet
-
-abstract class AggregateExpression extends Expression {
-  self: Product =>
 
-  /**
-   * Creates a new instance that can be used to compute this aggregate expression for a group
-   * of input rows/
-   */
-  def newInstance(): AggregateFunction
 
-  /**
-   * [[AggregateExpression.eval]] should never be invoked because [[AggregateExpression]]'s are
-   * replaced with a physical aggregate operator at runtime.
-   */
-  override def eval(input: Row = null): EvaluatedType =
-    throw new TreeNodeException(this, s"No function to evaluate expression. type: ${this.nodeName}")
-}
+/**
+ * This is from org.apache.hadoop.hive.ql.udf.generic.GenericUDAFEvaluator.Mode
+ * Just a hint for the UDAF developers which stage we are about to process,
+ * However, we probably don't want the developers knows so many details, here
+ * is just for keep consistent with Hive (when integrated with Hive), need to
+ * figure out if we have work around for that soon.
+ */
+@deprecated
+trait Mode
```

Comment:
`Mode` stuffs need to have access modifiers because they are used to represent the internal states of Aggregate. It doesn't matter for UDAF developers.